flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/legacy/metrics/MetricStore.java

```diff
@@ -57,6 +60,17 @@
 public class MetricStore {
     private static final Logger LOG = LoggerFactory.getLogger(MetricStore.class);
 
+    /**
+     * The set holds the names of the transient metrics which are no longer useful after a subtask
+     * reaches terminal state and shall be removed to avoid misleading users.
+     */
```

Comment:
Maybe add one more line: "Note that there may be other transient metrics, we currently only support cleaning these three"

Reply:
Added